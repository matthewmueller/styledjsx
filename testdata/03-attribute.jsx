export default () => {
  const Element = "div"
  return (
    <div>
      <div class="test" {...test.test} />
      <div class="test" {...test.test.test} />
      <div class="test" {...this.test.test} />
      <div data-test="test" />
      <div class="test" />
      <div class={"test"} />
      <div class={`test`} />
      <div class={`test${true ? " test2" : ""}`} />
      <div class={"test " + test} />
      <div class={["test", "test2"].join(" ")} />
      <div class={true && "test"} />
      <div class={test ? "test" : null} />
      <div class={test} />
      <div class={test && "test"} />
      <div class={test && test("test")} />
      <div class={undefined} />
      <div class={null} />
      <div class={false} />
      <div class={"test"} data-test />
      <div data-test class={"test"} />
      <div class={"test"} data-test="test" />
      <div class={"test"} {...props} />
      <div class={"test"} {...props} {...rest} />
      <div class={`test ${test ? "test" : ""}`} {...props} />
      <div class={test && test("test")} {...props} />
      <div class={test && test("test") && "test"} {...props} />
      <div class={test && test("test") && test2("test")} {...props} />
      <div {...props} class={"test"} />
      <div {...props} {...rest} class={"test"} />
      <div {...props} class={"test"} {...rest} />
      <div {...props} />
      <div {...props} {...rest} />
      <div {...props} data-foo {...rest} />
      <div {...props} class={"test"} data-foo {...rest} />
      <div {...{ id: "foo" }} />
      <div {...{ class: "foo" }} />
      <div {...{ class: "foo" }} class="test" />
      <div class="test" {...{ class: "foo" }} />
      <div {...{ class: "foo" }} {...bar} />
      <div {...{ class: "foo" }} {...bar} class="test" />
      <div class="test" {...{ class: "foo" }} {...bar} />
      <div class="test" {...{ class: props.class }} />
      <div class="test" {...{ class: props.class }} {...bar} />
      <div class="test" {...bar} {...{ class: props.class }} />
      <div class="test" {...bar()} />
      <Element />
      <Element class="test" />
      <Element {...props} />
      <style scoped>{"div { color: red }"}</style>
    </div>
  )
}
