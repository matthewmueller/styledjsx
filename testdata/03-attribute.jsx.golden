import Style from "styled-jsx";
export default () => {
  const Element = "div"
  return (
    <div class="jsx-2yZ49X">
      <div class="jsx-2yZ49X test" {...test.test} />
      <div class="jsx-2yZ49X test" {...test.test.test} />
      <div class="jsx-2yZ49X test" {...this.test.test} />
      <div data-test="test" class="jsx-2yZ49X" />
      <div class="jsx-2yZ49X test" />
      <div class={"jsx-2yZ49X test"} />
      <div class={`jsx-2yZ49X test`} />
      <div class={`jsx-2yZ49X test${true ? " test2" : ""}`} />
      <div class={"jsx-2yZ49X test " + test} />
      <div class={`jsx-2yZ49X ${["test", "test2"].join(" ")}`} />
      <div class={`jsx-2yZ49X ${true && "test"}`} />
      <div class={`jsx-2yZ49X ${test ? "test" : null}`} />
      <div class={`jsx-2yZ49X ${test}`} />
      <div class={`jsx-2yZ49X ${test && "test"}`} />
      <div class={`jsx-2yZ49X ${test && test("test")}`} />
      <div class={`jsx-2yZ49X ${undefined}`} />
      <div class={`jsx-2yZ49X ${null}`} />
      <div class={`jsx-2yZ49X ${false}`} />
      <div class={"jsx-2yZ49X test"} data-test />
      <div data-test class={"jsx-2yZ49X test"} />
      <div class={"jsx-2yZ49X test"} data-test="test" />
      <div class={"jsx-2yZ49X test"} {...props} />
      <div class={"jsx-2yZ49X test"} {...props} {...rest} />
      <div class={`jsx-2yZ49X test ${test ? "test" : ""}`} {...props} />
      <div class={`jsx-2yZ49X ${test && test("test")}`} {...props} />
      <div class={`jsx-2yZ49X ${test && test("test") && "test"}`} {...props} />
      <div class={`jsx-2yZ49X ${test && test("test") && test2("test")}`} {...props} />
      <div {...props} class={"jsx-2yZ49X test"} />
      <div {...props} {...rest} class={"jsx-2yZ49X test"} />
      <div {...props} class={"jsx-2yZ49X test"} {...rest} />
      <div {...props} class="jsx-2yZ49X" />
      <div {...props} {...rest} class="jsx-2yZ49X" />
      <div {...props} data-foo {...rest} class="jsx-2yZ49X" />
      <div {...props} class={"jsx-2yZ49X test"} data-foo {...rest} />
      <div {...{ id: "foo" }} class="jsx-2yZ49X" />
      <div {...{ class: "foo" }} class="jsx-2yZ49X" />
      <div {...{ class: "foo" }} class="jsx-2yZ49X test" />
      <div class="jsx-2yZ49X test" {...{ class: "foo" }} />
      <div {...{ class: "foo" }} {...bar} class="jsx-2yZ49X" />
      <div {...{ class: "foo" }} {...bar} class="jsx-2yZ49X test" />
      <div class="jsx-2yZ49X test" {...{ class: "foo" }} {...bar} />
      <div class="jsx-2yZ49X test" {...{ class: props.class }} />
      <div class="jsx-2yZ49X test" {...{ class: props.class }} {...bar} />
      <div class="jsx-2yZ49X test" {...bar} {...{ class: props.class }} />
      <div class="jsx-2yZ49X test" {...bar()} />
      <Element />
      <Element class="test" />
      <Element {...props} />
      <Style scoped>{`div.jsx-2yZ49X{color:red}`}</Style>
    </div>
  )
}
